tools/docgen/src/custom_markdown_documenter.ts

```diff
@@ -388,6 +388,8 @@ export class MarkdownDocumenter {
     pageContent = pageContent.replace(/<!-- -->/g, '');
     pageContent = pageContent.replace(/\\\*\\\*/g, '**');
     pageContent = pageContent.replace(/<b>|<\/b>/g, '**');
+    pageContent = pageContent.replace(/\{/g, '&#123;');
```

Comment:
Sadly this does not produce the desired result as we escape all the `{` and `}` for the page but we want to keep them in the context of `Signature` and `Example` else we get unusable code snippets:
![image](https://github.com/puppeteer/puppeteer/assets/34244704/b69246d7-d3d6-497a-b134-b3e7b9d3eb87)
![image](https://github.com/puppeteer/puppeteer/assets/34244704/2bc4d64c-c27f-45c5-8d45-da034e8de24f)
We would need to escape only in places where we inject plain text and keep the rest as is.

Reply:
Thanks for catching this; I changed the replacement from the html code to the backslash escape, so it will look like the following:
![image](https://github.com/puppeteer/puppeteer/assets/6616203/9c5aaca4-e3ac-4243-bc32-2e72200afa15)

![image](https://github.com/puppeteer/puppeteer/assets/6616203/ee3b867a-2993-4eaa-a3fa-bd375dba1a60)


And of course the same would be for a parameters table:
![image](https://github.com/puppeteer/puppeteer/assets/6616203/f6f448d9-bf09-4c73-8f5e-8cb766c2486a)

Reply:
This still creates the issue in a different way, please use the `Next` option from the version menu to verify. 

![image](https://github.com/puppeteer/puppeteer/assets/34244704/c8fdfc0d-ecf0-4454-a0e2-d8fbb8b62dfb)
![image](https://github.com/puppeteer/puppeteer/assets/34244704/46688274-ed15-45a5-a685-d5a08503f768)

Reply:
Alright, I changed the angle by extending the [api-formatter MarkdownEmitter](https://github.com/puppeteer/puppeteer/pull/11434/files#diff-81e1618806e847089a218d79340803b2981e837496c6cf850f16da8c51584720R96) and by overriding the `getEscapeText` and `getTableEscapedText` methods

It should be ok this way.
I'm sorry for the previous back and forth